.pre-commit-config.yaml

```diff
@@ -1,16 +1,42 @@
 # See https://pre-commit.com for more information
 # See https://pre-commit.com/hooks.html for more hooks
+exclude: versioneer.py
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v3.4.0
+    rev: v4.4.0
     hooks:
-      - id: trailing-whitespace
+      - id: check-builtin-literals
+      - id: check-case-conflict
+      - id: check-docstring-first
+      - id: check-executables-have-shebangs
+      - id: check-json
+      - id: check-toml
```

Comment:
I would remove checks that aren't even relevant to this repo, like json, toml, shebang, yaml, private-key, byteorder.

Reply:
I will get those cleaned up, that's just some no-effort random copypasta, maybe from some dask or anaconda related repo, at some point I must have stopped giving them much scrutiny.